libs/mctl/mctl.c

```diff
@@ -245,67 +301,38 @@ void mctl_deinit(void)
 	if (mctl_common.init) {
 		mctl_common.init = false;
 
-		for (i = 0; i < mctl_common.mNb; i++) {
-			if (mctl_common.pwmFiles[i] != NULL) {
-				fclose(mctl_common.pwmFiles[i]);
-			}
-		}
-
-		free(mctl_common.pwmFiles);
-		free(mctl_common.mThrottles);
+		free(mctl_common.motChannel);
 	}
 }
 
 
 int mctl_init(unsigned int motors, const char **motFiles)
 {
-	int id, cnt;
-	bool err;
+	int i;
 
-	if (motors == 0) {
+	if (motors == 0 || motors > ZYNQ7000_PWM_CHANNELS) {
 		return -1;
 	}
 	mctl_common.mNb = motors;
 
-	mctl_common.pwmFiles = calloc(mctl_common.mNb, sizeof(FILE *));
-	mctl_common.mThrottles = calloc(mctl_common.mNb, sizeof(float));
-	if (mctl_common.pwmFiles == NULL || mctl_common.mThrottles == NULL) {
-		free(mctl_common.pwmFiles);
-		free(mctl_common.mThrottles);
+	mctl_common.motChannel = calloc(mctl_common.mNb, sizeof(mctl_channel_t));
+	if (mctl_common.motChannel == NULL) {
+		free(mctl_common.motChannel);
+		printf("mctl: allocating channels error. Errno: %d\n", errno);
```

Comment:
```suggestion
	if (mctl_common.motChannel == NULL) {
		printf("mctl: allocating channels error. Errno: %d\n", errno);
```
`mctl_common.motChannel == NULL`

Reply:
Thanks! Missed that in adaptation commit. Removed.